src/simplex.pybind.h

```diff
@@ -70,27 +71,116 @@ void add_simplex_module(py::module& root) {
           "get_observables_from_errors",
           [](SimplexDecoder& self, const std::vector<size_t>& predicted_errors) {
             std::vector<bool> result(self.num_observables, false);
-            const auto& indices = self.get_flipped_observables(predicted_errors);
-            for (int index : indices) {
-              result[index] = true;
+            for (size_t ei : predicted_errors) {
+              for (int obs_index : self.errors[ei].symptom.observables) {
+                result[obs_index] = result[obs_index] ^ true;
+              }
             }
             return result;
           },
           py::arg("predicted_errors"))
       .def("cost_from_errors", &SimplexDecoder::cost_from_errors, py::arg("predicted_errors"))
       .def(
-          "decode",
-          [](SimplexDecoder& self, const std::vector<uint64_t>& detections) {
-            std::vector<bool> result(self.num_observables, false);
+          "decode_from_detection_events",
```

Comment:
My proposal was that the `decode` method takes a bitstring of detector outcomes as input (i.e. this function) but that we add a new method `decode_from_detection_events` which is implemented how `decode` _used_ to be defined.

```suggestion
          "decode",
```

In other words, `decode` should have arguments `(SimplexDecoder& self, const py::array_t<bool>& syndrome)` and `decode_from_detection_events` should have arguments `(SimplexDecoder& self, const std::vector<uint64_t>& detections)`

Reply:
@oscarhiggott Thank you. Done.